extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -691,6 +700,9 @@ private QueryDefinition initializeQueryDefAndState(final Closer closer)
         MSQControllerTask.isReplaceInputDataSourceTask(task)
     );
 
+    // propagate the controller's tags to the worker task
+    taskContextOverridesBuilder.put(DruidMetrics.TAGS, taskIdentitiesProvider.getTaskMetricTags(task));
+
```

Comment:
I don't think we should do this. This mixes user provided and system provided tags.

Is there a reason the taskIdentitiesProvider can not infer the system tags when it operates on the query_worker tasks?

Reply:
We need `taskIdentitiesProvider` to help identify the purpose of the task. For MSQ tasks, the purpose of what the task do is mainly inferred from `MSQControllerTask.spec`, `MSQWorkerTask` doesn't have this info, you can see `MSQWorkerTask` class itself it's very simple. Thus these info need to be passed in from the parent Controller task.